camel-chimera-rmlmapper/src/main/java/com/cefriel/rml/RmlConfigurator.java

```diff
@@ -167,11 +168,12 @@ static Mapper configure(Exchange exchange, RDFGraph graph, RecordsFactory factor
                 initializer = getInitializer(exchange, options);
 
             RDF4JRepository outputStore;
+            IRI namedGraph = graph.getNamedGraphs() != null ? graph.getNamedGraphs().get(0) : null;
```

Comment:
In this case, it is probably preferable not to provide the namedGraphs at all (null can be provided to the constructor). In theory, since the repository is "taken from the RDFGraph" the namedGraphs should be already set when the rmlmapper-cefriel is used within Chimera. The context is in the constructor for the standalone usage of the rmlmapper-cefriel (cf. https://github.com/cefriel/rmlmapper-cefriel/blob/42d71522e8ffb0e2ea56a77452452404c57f9a30/src/main/java/be/ugent/rml/store/RDF4JRepository.java#L53). Even if... since RML allows to generate quads (with the named graph for each triple specified) I am not sure why we have a context in the constructor of RDF4JRepository and ConcurrentRDF4JRepository.

Reply:
Adressed in 0d45fda